src/ShapeFile.php

```diff
@@ -326,8 +334,16 @@ public function getIndexFromDBFData(string $field, $value): int
      */
     private function loadDBFHeader(): array
     {
-        $DBFFile = fopen($this->getFilename('.dbf'), 'r');
+        if (! self::supportsDbase()) {
```

Comment:
I moved this block from `loadHeaders` so no matter where this method is called then we'll always check for support for `dbase`.